b_pull_Landsat_SRST_poi/py/gee_functions.py

```diff
@@ -0,0 +1,674 @@
+def csv_to_eeFeat(df, proj, wrs):
+  """Function to create an eeFeature from the location info
+
+  Args:
+      df: point locations .csv file with Latitude and Longitude
+      proj: CRS projection of the points
+
+  Returns:
+      ee.FeatureCollection of the points 
+  """
+  features=[]
+  for i in (df.index):
+    x,y = df.Longitude[i],df.Latitude[i]
+    latlong =[x,y]
+    loc_properties = {"system:index": str(df.id[i]), 
+    "id": str(df.id[i]),
+    "wrs": str(wrs)}
+    g=ee.Geometry.Point(latlong, proj) 
+    feature = ee.Feature(g, loc_properties)
+    features.append(feature)
+  ee_object = ee.FeatureCollection(features)
+  return ee_object
+
+
+def apply_scale_factors(image):
+  """ Applies scaling factors for Landsat Collection 2 surface reflectance 
+  and surface temperature products
+
+  Args:
+      image: one ee.Image of an ee.ImageCollection
+
+  Returns:
+      ee.Image with band values overwritten by scaling factors
+  """
+  opticalBands = image.select("SR_B.").multiply(0.0000275).add(-0.2)
+  thermalBands = image.select("ST_B.*").multiply(0.00341802).add(149.0)
+  return image.addBands(opticalBands, None, True).addBands(thermalBands, None,True)
+
+
+def dp_buff(image):
+  """ Buffer ee.FeatureCollection sites from csv_to_eeFeat by user-specified radius
+
+  Args:
+      image: ee.Image of an ee.ImageCollection
+
+  Returns:
+      ee.FeatureCollection of polygons resulting from buffered points
+  """
+  return image.buffer(ee.Number.parse(str(buffer)))
+
+
+def add_rad_mask(image):
+  """Mask out all pixels that are radiometrically saturated using the QA_RADSAT
+  QA band.
+
+  Args:
+      image: ee.Image of an ee.ImageCollection
+
+  Returns:
+      ee.Image with additional band called "radsat", where pixels with a value 
+      of 0 are saturated for at least one SR band and a value of 1 is not saturated
+  """
+  #grab the radsat band
+  satQA = image.select("radsat_qa")
+  # all must be non-saturated per pixel
+  satMask = satQA.eq(0).rename("radsat")
+  return image.addBands(satMask).updateMask(satMask)
```

Comment:
Minor thing here, if using .updateMask after .addBands, then the saturated pixels will be masked out so the additional bands will not have 0 values. You can consider switch the order of these two functions.

Reply:
If assuming 0 values in satQA band represents non-saturated pixels, then the logic is correct. basically any thing that equals one in r will be kept, which is what you want I think.  Because r is binary, you can simplify it to .updateMask(r).

If you remove .updateMask(satMask) here, I think it will not affect the line in the ref_pull... function so no need to change it to .updateMask(r.eq(0)).

Alternatively, you can make the satMask so that it equals satQA.eq(0) so in all your quality/flag bands, 0 means such condition does not exist (cloud, saturation, snow/ice) and 1 means it does. This way all the masking in ref_pull for these conditions should follow the format .updateMask(FLAG.eq(0))

Reply:
Awesome. Thank you. I'll revmoe the .updateMask(satMask), as it is redundant!